Realm/Realm/Exceptions/Sync/ErrorCode.cs

```diff
@@ -29,6 +38,11 @@ public enum ErrorCode
         /// </summary>
         Unknown = -1,
 
+        /// <summary>
+        /// A fatal error was encountered which prevents the completion of a client reset.
+        /// </summary>
+        AutoClientResetFailed_Cl = 132,
```

Comment:
I don't like adding this new value to the public enum. I suggest we add a new internal `ClientError` enum and add this value there. Then we add a new internal `SimulateError` overload that takes `ClientError`. Finally in `Session.ReportErrorForTesting` and in `realm_syncsession_report_error_for_testing` we add a new enum parameter (same enum in C# and C++) that indicates whether the error category is client or protocol.

Reply:
Unfortunately, `SimulateError` needs to be public in order to be visible outside the assembly so that it can be used by the tests. Because of this all of the types it uses in the signature need to be of the same visibility. Do you see any problem with `ClientError` being public?